### ISSUE TITLE
`strict-type-predicates` incompatible with type `unkown`

### ISSUE THREAD

**Issue by JounQin** [palantir/tslint#4357]
### Bug Report

- __TSLint version__: 5.11.0
- __TypeScript version__: 3.1.3
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
export const x = (params: unknown) => {
  if (typeof params === 'object') {
    return params
  }
}
```

with `tslint.json` configuration:

```json
{
  "extends": [
    "tslint:recommended",
    "tslint-react",
    "tslint-config-standard",
    "tslint-config-prettier",
    "tslint-plugin-prettier"
  ],
  "rules": {
    "member-access": [true, "no-public"],
    "no-floating-promises": false,
    "object-literal-sort-keys": false,
    "prettier": true
  }
}
```

#### Actual behavior
Error: `strict-type-predicates  Expression is always true.`

#### Expected behavior
No Eror

**Comment by JoshuaKGoldberg** [comment]
Agreed! Duplicate of #4107.
